lib/kubernetes-deploy/cluster_resource_discovery.rb

```diff
@@ -16,8 +16,26 @@ def crds
       end
     end
 
+    def global_resource_names
+      @globals ||= fetch_globals.map { |g| g["kind"] }
+    end
+
     private
 
+    def fetch_globals
+      raw, _, st = kubectl.run("api-resources", "--namespaced=false", output: "wide", attempts: 5)
+      if st.success?
```

Comment:
I _think_ I understand what you're doing here, but this code is so dense that it is hard to follow. Can you please expand it to do fewer things per line. stick to descriptive variables, and maybe add a comment or two?

I'm also thinking this warrants a few unit tests showing it working with different column widths.